subprojects/openssl.wrap

```diff
@@ -8,4 +8,4 @@ patch_directory = openssl
 [provide]
 libcrypto = libcrypto_dep
 libssl = libssl_dep
-openssl = openssl_dep
+openssl = libssl_dep
```

Comment:
libssl and openssl both uses libssl_dep, that does not seems right. I think you need to `openssl_dep = declare_dependency(dependencies: [libssl_dep, libcrypto_dep])`, no?

Reply:
Maybe, but `libssl` in turn depends on `libcrypto`, so should I still have a separate thing for `openssl`?

Reply:
See, openssl.pc contains: `Requires: libssl libcrypto`. So "openssl" dep really is just a meta dependency that has ssl and crypto.

Reply:
> Maybe, but `libssl` in turn depends on `libcrypto`, so should I still have a separate thing for `openssl`?

ah, ok, the the pc file does that just to specify that libcrypto is a public dep and not private dep, which makes no difference in meson...  Still I think it would make sense to have it explicit. You do it yourself for openssl_cli: `dependencies: dependencies + [libcrypto_dep, libssl_dep],`. But that's just a small nitpicking then.

Reply:
A declare_dependency on a shared_library('ssl') will add recursive dependencies on the shared_library('crypto') implementation detail?